### PR TITLE
global: application scaffolding

### PR DESCRIPTION
**Pull request by kprzerwa**
* ils theme
* minimal landing page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+  "name": "ils-app",
```
Comment (ntarocco) [inline package.json:2]:
```suggestion
  "name": "invenio-app-ils-ui",
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  "version": "0.1.0",
+  "private": true,
```
Comment (ntarocco) [inline package.json:4]: why this?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "build": "craco build",
+    "test": "craco test",
+    "test:ci": "CI=true craco test",
+    "eject": "craco eject",
+    "format": "prettier --config ./.prettierrc --ignore-path ./.prettierignore --write \"**/*.js\""
+  },
+  "dependencies": {
+    "axios": "^0.19.2",
+    "formik": "^2.1.4",
+    "less": "^3.11.1",
```
Comment (ntarocco) [inline package.json:16]: I think less is a dev dependency?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  "private": true,
+  "scripts": {
+    "start": "craco start",
+    "build": "craco build",
+    "test": "craco test",
+    "test:ci": "CI=true craco test",
+    "eject": "craco eject",
+    "format": "prettier --config ./.prettierrc --ignore-path ./.prettierignore --write \"**/*.js\""
+  },
+  "dependencies": {
```
Comment (ntarocco) [inline package.json:13]: all these ones should be `peerDeps`, we should have 0 deps.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "axios": "^0.19.2",
+    "formik": "^2.1.4",
+    "less": "^3.11.1",
+    "less-loader": "^5.0.0",
+    "lodash": "^4.17.15",
+    "luxon": "^1.23.0",
+    "path": "^0.12.7",
+    "prop-types": "^15.7.2",
```
Comment (ntarocco) [inline package.json:21]: devDep?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "format": "prettier --config ./.prettierrc --ignore-path ./.prettierignore --write \"**/*.js\""
+  },
+  "dependencies": {
+    "axios": "^0.19.2",
+    "formik": "^2.1.4",
+    "less": "^3.11.1",
+    "less-loader": "^5.0.0",
+    "lodash": "^4.17.15",
+    "luxon": "^1.23.0",
+    "path": "^0.12.7",
```
Comment (ntarocco) [inline package.json:20]: I am not sure about this `path`, to be reviewed

File: package.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    "prop-types": "^15.7.2",
+    "qs": "^6.9.3",
+    "react": "^16.13.1",
+    "react-app-polyfill": "^1.0.6",
+    "react-copy-to-clipboard": "^5.0.2",
+    "react-dom": "^16.13.1",
+    "react-overridable": "^0.0.2",
+    "react-redux": "^7.2.0",
+    "react-router-dom": "^5.1.2",
+    "react-scripts": "^3.4.1",
```
Comment (ntarocco) [inline package.json:30]: devDep

File: package.json [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    "react-scroll": "^1.7.16",
+    "react-searchkit": "^0.19.0",
+    "react-show-more": "^2.0.0",
+    "react-tagcloud": "^2.0.0",
+    "redux": "^4.0.5",
+    "redux-devtools-extension": "^2.13.8",
```
Comment (ntarocco) [inline package.json:36]: this I am not sure if only devDep

File: package.json [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    "@types/jest": "^24.9.1",
+    "axios-mock-adapter": "^1.18.1",
+    "enzyme": "^3.11.0",
+    "enzyme-adapter-react-16": "^1.15.2",
+    "enzyme-to-json": "^3.4.4",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-prettier": "^3.1.3",
+    "eslint-plugin-react": "^7.20.0",
+    "lint-staged": "^9.5.0",
+    "node-sass": "^4.14.0",
```
Comment (ntarocco) [inline package.json:55]: do we need `node-sass`?

File: public/index.html [hunk 10]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
```
Comment (ntarocco) [inline public/index.html:1]: what about if we have: 
```
src/
    demo/
    lib/
```
wouldn't it better?

File: src/index.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Provider } from "react-redux";
+import "react-app-polyfill/ie11"; // For IE 11 support
+import App from "./standalone/App";
+import store from "./store";
+import "semantic-ui-less/semantic.less";
+
+ReactDOM.render(
+  <Provider store={store}>
```
Comment (ntarocco) [inline src/index.js:10]: I have the feeling that the `Provider` should be inside the lib itselfs.

File: src/lib/components/ui/ILSFooter/ILSFooter.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
+import { getStaticPageByName } from "@config/uiConfig";
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Overridable from "react-overridable";
+import { Link } from "react-router-dom";
+import { Container, Grid, Header, List } from "semantic-ui-react";
+
+const OverridableFooter = props => {
```
Comment (ntarocco) [inline src/lib/components/ui/ILSFooter/ILSFooter.js:8]: not sure I understand the overridable pattern, you are basically defining 2 overridable templates that are the same.
It should be:
```
class Footer extends Component {
  render() {
    return <Overridable id="Footer.container" {...this.props}>...</>;
  }
}

export default Overridable.component("Footer", Footer);
```

this is valid for all the components here

```

File: src/lib/components/ui/ILSMenu/index.js [hunk 13]
```diff
@@ -0,0 +1,20 @@
+import { connect } from "react-redux";
+import ILSMenuComponent from "./ILSMenu";
+// import { addNotification } from "@components/Notifications/state/actions";
```
Comment (ntarocco) [inline src/lib/components/ui/ILSMenu/index.js:3]: Very dangerous to have code comment out IMHO, we will forget some. That's why I was proposing to stop working on UI and to move a block of components.

File: src/lib/components/ui/NotFound/NotFound.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Grid, Button, Icon } from 'semantic-ui-react';
+
+export class NotFound extends Component {
```
Comment (ntarocco) [inline src/lib/components/ui/NotFound/NotFound.js:5]: Overridable?

File: src/lib/components/ui/SearchBar/QueryBuildHelper.js [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+QueryBuildHelper.propTypes = {
+  fields: PropTypes.array.isRequired,
+  currentQueryString: PropTypes.string.isRequired,
+  updateQueryString: PropTypes.func.isRequired,
+  size: PropTypes.number
+};
+
+QueryBuildHelper.defaultProps = {
+  // size: invenioConfig.defaultResultsSize
```
Comment (ntarocco) [inline src/lib/components/ui/SearchBar/QueryBuildHelper.js:76]: ?

File: src/lib/components/ui/SearchBar/SearchBar.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+    );
+  }
+}
+
+SearchBar.propTypes = {
+  currentQueryString: PropTypes.string,
+  onInputChange: PropTypes.func.isRequired,
+  executeSearch: PropTypes.func.isRequired,
+  placeholder: PropTypes.string.isRequired,
+  queryHelperFields: PropTypes.array,
```
Comment (ntarocco) [inline src/lib/components/ui/SearchBar/SearchBar.js:72]: what is not required should have a default I think

File: src/lib/components/ui/ILSFooter/ILSFooter.js [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+
+ILSFooter.propTypes = {
+  staticPages: PropTypes.arrayOf(
+    PropTypes.shape({
+      name: PropTypes.string.isRequired,
+      route: PropTypes.string.isRequired,
+      apiURL: PropTypes.string.isRequired
+    })
+  )
+};
```
Comment (ntarocco) [inline src/lib/components/ui/ILSFooter/ILSFooter.js:71]: missing default

File: src/lib/components/ui/SearchBar/__tests__/SearchBar.test.js [hunk 18]
```diff
@@ -0,0 +1,121 @@
+import React from "react";
```
Comment (ntarocco) [inline src/lib/components/ui/SearchBar/__tests__/SearchBar.test.js:1]: please remove the `__tests__` folder

File: src/lib/config/uiConfig.js [hunk 19]
```diff
@@ -0,0 +1,19 @@
+import _find from "lodash/find";
```
Comment (ntarocco) [inline src/lib/config/uiConfig.js:1]: for the config overriding, I think we can discuss IRL

File: src/lib/helpers/propTypes.js [hunk 20]
```diff
@@ -0,0 +1,54 @@
+/**
+ * until https://github.com/facebook/prop-types/issues/263 fixed
```
Comment (ntarocco) [inline src/lib/helpers/propTypes.js:2]: Can you also add from where you took it?

File: src/semantic-ui/site/collections/form.overrides [hunk 21]
```diff
@@ -0,0 +1,68 @@
+/*******************************
```
Comment (ntarocco) [inline src/semantic-ui/site/collections/form.overrides:1]: theming to be discussed later on IRL

File: src/standalone/App.js [hunk 22]
```diff
@@ -0,0 +1,38 @@
+import { NotFound } from "../lib/components/ui";
```
Comment (ntarocco) [inline src/standalone/App.js:1]: I see there is an App here, then you should remove the App from the lib, it should not have an app.
IMHO, I would prefer `demo` or similar to `standalone`, not sure I understand what that means.

File: src/standalone/App.js [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+import React, { Component } from "react";
+import { Router, Route, Switch } from "react-router-dom";
+import { FrontSite } from "../lib/routes/";
+import history from "../lib/history";
```
Comment (ntarocco) [inline src/standalone/App.js:5]: you should be able to use aliases too

File: package.json (hunk 1 above)
Comment (kprzerwa) [reply package.json:2]: 👍

File: package.json (hunk 2 above)
Comment (kprzerwa) [reply package.json:4]: copied from ILS, I guess someone wanted to prevent from publishing?

File: src/lib/config/uiConfig.js (hunk 19 above)
Comment (kprzerwa) [reply src/lib/config/uiConfig.js:1]: yes I didn't start with this yet, I copied for now only to make things work.  But yes it needs to be done a bit different

File: src/lib/helpers/propTypes.js (hunk 20 above)
Comment (kprzerwa) [reply src/lib/helpers/propTypes.js:2]: 👍

File: src/standalone/App.js (hunk 23 above)
Comment (kprzerwa) [reply src/standalone/App.js:5]: 👍

**Pull request closed without merging**